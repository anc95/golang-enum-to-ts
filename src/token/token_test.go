package token

import (
	"os"
	"path"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestNormal(t *testing.T) {
	wd, _ := os.Getwd()

	a := path.Join(wd, "../test-cases/normal.go")
	parser := NewParser(a)
	result := parser.Parse()

	snaps.MatchSnapshot(t, result)
}
