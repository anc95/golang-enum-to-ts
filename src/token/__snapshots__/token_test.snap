
[TestNormal - 1]
[]token.Token{
    {
        Value: "package",
        Type:  20,
        Start: {0, 6},
        End:   {0, 0},
    },
    {
        Value: "some",
        Type:  20,
        Start: {0, 11},
        End:   {0, 0},
    },
    {
        Value: "type",
        Type:  1,
        Start: {2, 3},
        End:   {0, 0},
    },
    {
        Value: "Status",
        Type:  20,
        Start: {2, 10},
        End:   {0, 0},
    },
    {
        Value: "int",
        Type:  4,
        Start: {2, 14},
        End:   {0, 0},
    },
    {
        Value: "const",
        Type:  3,
        Start: {4, 5},
        End:   {0, 0},
    },
    {
        Value: "(",
        Type:  18,
        Start: {4, 6},
        End:   {0, 0},
    },
    {
        Value: "Todo",
        Type:  20,
        Start: {5, 4},
        End:   {0, 0},
    },
    {
        Value: "Status",
        Type:  20,
        Start: {5, 11},
        End:   {0, 0},
    },
    {
        Value: "=",
        Type:  9,
        Start: {5, 13},
        End:   {0, 0},
    },
    {
        Value: "iota",
        Type:  6,
        Start: {5, 18},
        End:   {0, 0},
    },
    {
        Value: "Done",
        Type:  20,
        Start: {6, 4},
        End:   {0, 0},
    },
    {
        Value: ")",
        Type:  19,
        Start: {7, 0},
        End:   {0, 0},
    },
}
---
